libs/types/src/solana/mod.rs

```diff
@@ -95,3 +96,28 @@ pub struct PrioritizationFee {
     #[serde(rename = "prioritizationFee")]
     pub prioritization_fee: MicroLamport,
 }
+
+/// A vector with a maximum capacity.
+#[derive(Debug, Clone, Deserialize, Serialize, CandidType, PartialEq, Default)]
+#[serde(try_from = "Vec<T>")]
```

Comment:
shouldn't we have also `into` `Vec<T>` for serialization?

```suggestion
#[serde(try_from = "Vec<T>", into="Vec<T>")]
```

Reply:
> Hm I added some tests to make sure that encoding/decoding round trips with Candid work and it seems this is not necessary.

I think that's expected because we only use those Candid types to deserialize user inputs, but the Candid encoding of MaxVecLength<T> would result in different bytes than the Candid encoding of the underlying Vec<T>, which is I think unexpected. To test this you do a test similar to this [one](https://github.com/dfinity/sol-rpc-canister/blob/1744026c3ab9ddd5613ecece68b6e2259946b0ee/libs/types/src/solana/tests.rs#L63).

> adding this forces us to add the Clone restriction on the generic type T

I'm not following why `Clone`, the bound should be `T: Serialize`, no?